src/Core/MakeModelMenu.elm

```diff
@@ -0,0 +1,350 @@
+                            RemoteData.update (processRemoteData .factoryOrderAvailable) response
+
+                        "Stock" ->
+                            RemoteData.update (processRemoteData .stockAvailable) response
+
+                        "LeaseDeals" ->
+                            RemoteData.update (processRemoteData .leaseDealsAvailable) response
+
+                        _ ->
+                            ( RemoteData.Loading, Cmd.none )
+
+                newState =
+                    case model.state of
+                        ModelSelection make _ ->
+                            ModelSelection make sortedFilteredResponse
+
+                        _ ->
+                            model.state
+            in
+                ( { model | state = newState }, Cmd.none )
+
+        ModalBackClicked ->
+            let
+                newState =
+                    MakeSelection RemoteData.Loading
+            in
+                ( { model | state = newState }, getAvailableMakes (makesApiUrl model.apiEndpointUrl) )
+
+
+{-| Placeholder
+-}
+processRemoteData :
+    (MenuItem a -> Bool)
+    -> List (MenuItem a)
+    -> ( List (MenuItem a), Cmd msg )
+processRemoteData field items =
+    let
+        sortField item =
+            item.name |> String.toUpper
+    in
+        ( items |> List.filter field |> List.sortBy sortField, Cmd.none )
+
+
+{-| Placeholder
+-}
+listView : (a -> List (Html msg)) -> WebData (List a) -> Html msg
+listView itemView itemsRemoteData =
+    case itemsRemoteData of
+        RemoteData.Success items ->
+            ul [ class "makes-models-menu__list list-unstyled" ]
+                (items
+                    |> List.map
+                        (\item ->
+                            li
+                                [ class "makes-models-menu__list-item" ]
+                                (itemView item)
+                        )
+                )
+
+        _ ->
+            spinnerView
+
+
+{-| Placeholder
+-}
+modalMakesView : WebData (List Make) -> Html Msg
+modalMakesView makesRemoteData =
+    let
+        makeView =
+            (\make ->
+                [ span [ class "makes-models-menu__link makes-models-menu__make", Html.Events.onClick (MakeSelected make) ]
+                    [ icon ("makes/" ++ String.join "_" (String.split "-" (String.toLower make.slug))) { size = "large", colour = "grey", colouring = "filled" }
+                    , div [ class "makes-models-menu__make" ] [ text make.name ]
+                    ]
+                ]
+            )
+    in
+        listView makeView makesRemoteData
+
+
+{-| Placeholder
+-}
+modalModelsView : Core.Data.Make.Make -> String -> Erl.Url -> WebData (List Core.Data.Model.Model) -> Html Msg
+modalModelsView make location baseLinkUrl modelsRemoteData =
+    let
+        makeModelUrl =
+            (\model ->
+                location
+                    |> Regex.replace All (Regex.regex "www.") (\_ -> "quotes.")
```

Comment:
`quotes.` is not i18n-friendly - in germany quotes is under the domain `angebote.`